### PR TITLE
fix/eth_getLogs

### LINKED ISSUE #33: Invalid topics format in eth_getLogs
Just tested the return of `eth_getLogs` RPC (below): 
```bash
{
            "address": "0xa48bc38ef6ea769792bd095a53f3cc83c8391047",
            "blockHash": "0x01e7c7dcc8c7405e2e14dc78932140ef1e49bb9d08702f97bd84a0ae8f5352e0",
            "blockNumber": "0x3a97426",
            "data": "0x0000000000000000000000006a33382de9f73b846878a57500d055b981229ac400000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000064000000000000000000000000000000000000000000000000000000000000002a30784134386243333845463665613736393739324264303935413533463343633833433833393130343700000000000000000000000000000000000000000000",
            "logIndex": "0x0",
            "removed": false,
            "topics": "{\"\\\\x985c84346e3bff1de514922d1bf88270b4a50c525b86443aeeb1441c842e3815\",\"\\\\x0000000000000000000000000000000000000000000000000000000000000001\"}",
            "transactionHash": "0x5b760c2ee670b8100e8716da75123a85ff3cd2790a371edeca40ed90db84e6aa",
            "transactionIndex": "0x0"
        }
```

The topics field should be an array of 32 bytes as follows:

```bash
{
            "address": "0xa48bc38ef6ea769792bd095a53f3cc83c8391047",
            "blockHash": "0x01e7c7dcc8c7405e2e14dc78932140ef1e49bb9d08702f97bd84a0ae8f5352e0",
            "blockNumber": "0x3a97426",
            "data": "0x0000000000000000000000006a33382de9f73b846878a57500d055b981229ac400000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000064000000000000000000000000000000000000000000000000000000000000002a30784134386243333845463665613736393739324264303935413533463343633833433833393130343700000000000000000000000000000000000000000000",
            "logIndex": "0x0",
            "removed": false,
            "topics": [
                "0x985c84346e3bff1de514922d1bf88270b4a50c525b86443aeeb1441c842e3815",
                "0x0000000000000000000000000000000000000000000000000000000000000001"
            ],
            "transactionHash": "0x5b760c2ee670b8100e8716da75123a85ff3cd2790a371edeca40ed90db84e6aa",
            "transactionIndex": "0x0"
        }
```

Please refer to the [reference documentation](https://docs.alchemy.com/alchemy/guides/eth_getlogs) for the result of calling `eth_getLogs`.

### PR DESCRIPTION
**Pull request by 0x3bfc**
closes #33 
- [x] Fix the index in `compileTopics`
- [x] Return topics as an array of strings instead of hash array (not RPC friendly format).

### CODE REVIEW

File: src/topics.ts [hunk 1]
```diff
@@ -33,3 +33,15 @@ export function compileTopics(
...
+  return input
+  .split(',')
+  .map(
+    (t: string) => 
+    t.replace("{\"\\\\", '0')
+    .replace("\"\\\\", "0")
+    .replace("\"}", "")
+    .replace("\"", "")
+  );
+}
```
Comment (artob) [inline src/topics.ts:47]: If run through the linter, it should complain about the missing trailing newline

File: src/topics.ts [hunk 2]
```diff
@@ -33,3 +33,15 @@ export function compileTopics(
...
+
+export function toStringArray(input: string): Array<string>{
+  return input
+  .split(',')
+  .map(
+    (t: string) => 
+    t.replace("{\"\\\\", '0')
+    .replace("\"\\\\", "0")
+    .replace("\"}", "")
+    .replace("\"", "")
```
Comment (artob) [inline src/topics.ts:45]: This seems pretty fragile, and I'm not clear on why it's needed. Could you elaborate a bit, @0x3bfc?

Comment (0x3bfc) [reply src/topics.ts:45]: Totally agree! 
Will try to find a way to convert the buffer array from postgress to an array of hex values.

File: src/indexer.ts [hunk 3]
```diff
@@ -232,7 +232,6 @@ export class Indexer {
         ? event.topics.map((topic) => topic.toBytes())
         : null,
     });
-
```
Comment (antonpaisov) [inline src/indexer.ts:235]: no need to push changes in files that aren't really affected 🙂

File: src/topics.ts [hunk 4]
```diff
@@ -32,4 +32,4 @@ export function compileTopics(
 // Duplicated here because of https://github.com/kulshekhar/ts-jest/issues/970
 function hexToBytes(input: string): Uint8Array {
   return Buffer.from(input.substring(2), 'hex');
-}
+}
```
Comment (antonpaisov) [inline src/topics.ts:35]: there should be a newline in the end of the file

**Review by antonpaisov — CHANGES_REQUESTED** [review]
LGTM generally with a few minor comments.
We need to keep this query in mind when we do integration testing and add some proper tests.

**Review by antonpaisov — APPROVED** [review]

**Comment by 0x3bfc** [comment]
> there should be a newline in the end of the file

I think we need to enforce the linting issue reporting in the CI. Opening an issue [here](https://github.com/aurora-is-near/aurora-relayer/issues/37)

File: src/servers/database.ts [hunk 5]
```diff
@@ -357,7 +357,7 @@ export class DatabaseServer extends SkeletonServer {
         't.hash AS "transactionHash"',
         'e.index AS "logIndex"',
         't.to AS "address"',
-        'e.topics AS "topics"',
+        'string_to_array(concat(\'0x\',encode(e.topics[1], \'hex\'), \',\', \'0x\', encode(e.topics[2], \'hex\'), \',\', \'0x\', encode(e.topics[3], \'hex\'), \',\', \'0x\', encode(e.topics[4], \'hex\')), \',\') AS "topics"',
```
Comment (mfornet) [inline src/servers/database.ts:360]: What is the difference between previous line and this line.

Also, I see there are not tests for this repository. Did you manually confirm this changes fixes the issue.

**Review by mfornet — COMMENTED** [review]
This is my first review to eth_getLogs

File: src/servers/database.ts (hunk 5 above)
Comment (0x3bfc) [reply src/servers/database.ts:360]: > What is the difference between previous line and this line.

The problem was mentioned in #33 

> Also, I see there are not tests for this repository. 

To do that, I've to test it against working database. the mocking might take long time for multiple components. 

> Did you manually confirm this changes fixes the issue.

Yes, I've tested this on the testnet including other edge cases.

**Review by artob — APPROVED** [review]

**Pull request merged**